custom_components/tuya_local/devices/tcl_airmax_airconditioner.yaml

```diff
@@ -0,0 +1,305 @@
+name: Air conditioner
+products:
+  - id: hw50w7qvxluhslkk
+    manufacturer: TCL
+    model: AirMax
+entities:
+  - entity: climate
+    translation_only_key: aircon_extra
+    dps:
+      - id: 1
+        type: boolean
+        name: hvac_mode
+        mapping:
+          - dps_val: false
+            value: "off"
+          - dps_val: true
+            constraint: mode
+            conditions:
+              - dps_val: cold
+                value: cool
+              - dps_val: hot
+                value: heat
+              - dps_val: wet
+                value: dry
+              - dps_val: wind
+                value: fan_only
+              - dps_val: auto
+                value: heat_cool
+      - id: 2
+        type: integer
+        name: temperature
+        unit: F
+        range:
+          min: 630
+          max: 880
+        mapping:
+          - scale: 10
+            step: 10
+      - id: 3
+        type: integer
+        name: current_temperature
+        mapping:
+          - dps_val: 16
+            value: 61
+          - dps_val: 17
+            value: 63
+          - dps_val: 18
+            value: 64
+          - dps_val: 19
+            value: 66
+          - dps_val: 20
+            value: 68
+          - dps_val: 21
+            value: 70
+          - dps_val: 22
+            value: 72
+          - dps_val: 23
+            value: 73
+          - dps_val: 24
+            value: 75
+          - dps_val: 25
+            value: 77
+          - dps_val: 26
+            value: 79
+          - dps_val: 27
+            value: 81
+          - dps_val: 28
+            value: 82
+          - dps_val: 29
+            value: 84
+          - dps_val: 30
+            value: 86
+          - dps_val: 31
+            value: 88
+          - dps_val: 32
+            value: 90
+      - id: 4
+        type: string
+        name: mode
+        hidden: true
+      - id: 5
+        type: string
+        name: fan_mode
+        mapping:
+          - dps_val: "strong"
+            value: strong
+          - dps_val: "high"
+            value: high
+          - dps_val: "mid_high"
+            value: medhigh
+          - dps_val: "mid"
+            value: medium
+          - dps_val: "mid_low"
+            value: medlow
+          - dps_val: "low"
+            value: low
+          - dps_val: "mute"
+            value: quiet
+          - dps_val: "auto"
+            value: auto
+      - id: 18
+        type: integer
+        name: current_humidity
+      - id: 110
+        type: bitfield
+        name: flags
+      - id: 113
+        type: string
+        name: swing_mode
+        mapping:
+          - dps_val: "0"
+            value: "off"
+          - dps_val: "1"
+            value: "on"
+          - hidden: true
```

Comment:
I think this can cover the full range of vertical swing modes so the select entity below is not needed. Unlike hvac_mode, swing_mode is not limited to predefined modes, so you can use  Upper and Lower here (and change "on" to Full if you think it would be clearer)